libpod/container_attach.go

```diff
@@ -0,0 +1,144 @@
+package libpod
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/docker/docker/pkg/term"
+	"github.com/kubernetes-incubator/cri-o/utils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"k8s.io/client-go/tools/remotecommand"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"net"
+)
+
+/* Sync with stdpipe_t in conmon.c */
+const (
+	AttachPipeStdin  = 1
+	AttachPipeStdout = 2
+	AttachPipeStderr = 3
+)
+
+// attachContainerSocket connects to the container's attach socket and deals with the IO
+func (c *Container) attachContainerSocket(resize <-chan remotecommand.TerminalSize, noStdIn bool, detachKeys []byte) error {
+	inputStream := os.Stdin
```

Comment:
Noted it earlier, but can we pass these in as arguments instead of assuming os.Stdin/out/err?

Reply:
I think we should only be passing in stdin for this. Althought I would be game to change the stdin to a be a device or nil.

Reply:
I can see cases where we just want to pass in buffers for stdin and stdout to run a container headless and capture the output. Unit tests, for example - pass in a bytes.Buffer for stdout/stderr to capture the output of the container and check the buffers to make sure you have the right output